src/server/list_family.cc

```diff
@@ -146,23 +173,43 @@ class BPopPusher {
 
 // Called as a callback from MKBlocking after we've determined which key to pop.
 std::string OpBPop(Transaction* t, EngineShard* shard, std::string_view key, ListDir dir) {
+  DVLOG(2) << "popping from " << key << " " << t->DebugId();
+
   auto& db_slice = shard->db_slice();
   auto it_res = db_slice.Find(t->GetDbContext(), key, OBJ_LIST);
+
+  if (!it_res) {
+    auto messages = debugMessages.All();
+    cerr << "CRASH REPORT" << endl;
+    cerr << "key: " << key << " tx: " << t->DebugId() << "\n";
+    cerr << "===\n";
```

Comment:
Any reason not to use `LOG(ERROR)`? It can print additional information, like thread id, time, etc.
`ERROR` (and `FATAL`) messages are by default also printed to `stderr`, so no need for extra flags to see them.

Reply:
Its easier to copy those 100 lines without clutter. We don't need the thread-id because its all on the same thread. Time maybe useful somehow. Either way I my focus is on readability and detectability

Reply:
The thing is, by outputting directly to `stderr`, you're effectively skipping the log file, which might be what the users actually have..